static/js/utils.js

```diff
@@ -0,0 +1,16 @@
+/**
```

Comment:
would it make sense to do this kind of data cleaning before the listener is called? not sure how possible this would be, but it would be nice if the listeners didn't have to worry about this (and perform the same canonicalization multiple times)

Reply:
ya, I can experiment with this. We could have an args param in the listeners which indicates the data types of the callback args so that we can canonicalize accordingly

Reply:
I moved canonicalization to the runtime config itself